cmd/datamon/cmd/cli_workshop_test.go

```diff
@@ -0,0 +1,494 @@
+	require.Equal(t, "signal: killed", err.Error(), "cmd exit with killed error")
+
+	// downloading the bundle
+	pathToDownload, _ := ioutil.TempDir("", "mmd-")
+	defer os.RemoveAll(pathToDownload)
+
+	cmd, _ = testCommand(t, false, false, targetBinary,
+		"bundle",
+		"download",
+		"--repo", repo,
+		"--label", label,
+		"--destination", pathToDownload,
+	)
+	output, err = cmd.CombinedOutput()
+	require.NoErrorf(t, err, "unexpected error, got: %s", string(output))
+
+	err = filepath.Walk(pathToDownload, func(pth string, info os.FileInfo, _ error) error {
+		rel, _ := filepath.Rel(pathToDownload, pth)
+		if info.IsDir() || model.IsGeneratedFile(rel) {
+			return nil
+		}
+		name := filepath.Base(pth)
+		require.Contains(t, files, name)
+		files[name].downloaded = true
+		assert.Equal(t, files[name].size, int(info.Size()))
+		b, erf := ioutil.ReadFile(pth)
+		require.NoError(t, erf)
+		assert.EqualValues(t, files[name].data, b)
+		return nil
+	})
+	require.NoError(t, err)
+	for name, file := range files {
+		assert.Truef(t, file.downloaded, "expected file %s to be downloaded", name)
+	}
+
+	// downloading some files
+	pathToSomeDownload, _ := ioutil.TempDir("", "mmd-")
+	defer os.RemoveAll(pathToSomeDownload)
+
+	var picked string
+	for n := range files {
+		picked = n
+		break
+	}
+
+	cmd, _ = testCommand(t, false, false, targetBinary,
+		"bundle",
+		"download",
+		"--repo", repo,
+		"--label", label,
+		"--destination", pathToSomeDownload,
+		"--name-filter", picked[:8]+".+",
+	)
+	output, err = cmd.CombinedOutput()
+	require.NoErrorf(t, err, "unexpected error, got: %s", string(output))
+
+	err = filepath.Walk(pathToSomeDownload, func(pth string, info os.FileInfo, _ error) error {
+		rel, _ := filepath.Rel(pathToSomeDownload, pth)
+		if info.IsDir() || model.IsGeneratedFile(rel) {
+			return nil
+		}
+		name := filepath.Base(pth)
+		require.Equal(t, picked, name)
+		assert.Equal(t, files[name].size, int(info.Size()))
+		b, erf := ioutil.ReadFile(pth)
+		require.NoError(t, erf)
+		assert.EqualValues(t, files[name].data, b)
+		return nil
+	})
+	require.NoError(t, err)
+
+	// no test here for web (see pkg/web)
+	// no test here for upgrade (see self_upgrade_test.go)
+}
+
+func makeTestWorkshopBundle(t testing.TB, pathToData string) map[string]*testFile {
+	files := make(map[string]*testFile, 10)
+	for i := 0; i < 9; i++ {
+		name := internal.RandStringBytesMaskImprSrc(10)
+		data := internal.RandBytesMaskImprSrc(100)
+		err := ioutil.WriteFile(filepath.Join(pathToData, name), data, 0644)
+		require.NoError(t, err)
+		files[name] = &testFile{data: data, size: 100}
+	}
+	name := internal.RandStringBytesMaskImprSrc(10)
+	data := []byte{} // empty file
+	err := ioutil.WriteFile(filepath.Join(pathToData, name), data, 0644)
+	require.NoError(t, err)
+	files[name] = &testFile{data: data, size: 0}
+	return files
+}
+
+func TestWorkshopBrewInstall(t *testing.T) {
+	if runtime.GOOS == "linux" { // TODO: don't know where brew installs things on OSX
```

Comment:
> where brew installs things on OSX

```
~% whence datamon2
/usr/local/bin/datamon2
~% ls -l /usr/local/bin/datamon2
lrwxr-xr-x  1 ransomwilliams  admin  42 Jan 14 10:56 /usr/local/bin/datamon2 -> ../Cellar/datamon/2.0.0-beta6/bin/datamon2
```

of course, so `/usr/local/Cellar/`?

except i suspect the question is more nuanced that i'm representing it.

Reply:
@ransomw1c well I don't know really how brew runs on MacOS. I've installed it on linux (specifically for datamon, since I am not a regular user of brew) and I designed this test to be executed by CI, which runs brewlinux. I think this is _reasonably_ close to the targeted audience. However, I cannot tell how this works on Mac (e.g. if you run go -test locally) hence I disabled the special path resolution we have on linux distros (where basically brew installs its wares on some home directory, and not in /usr). If this the command is by default in the path on mac, then my code should work also on mac.